src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -197,12 +198,35 @@ def __enter__(self):
         # Starting a new process in non-fork mode requires to allocate memory. Determine the maximum number of processes
         # based on available memory to avoid memory bursting.
         if not self._use_fork:
-            available_max_worker_count = get_available_max_worker_count()
-            self._n_process = min(self._worker_count, self._nlines, available_max_worker_count)
-            bulk_logger.info(f"Not using fork, process count: {self._n_process}")
+            estimated_available_worker_count = get_available_max_worker_count()
+            if self._use_default_worker_count:
+                self._n_process = min(self._worker_count, self._nlines, estimated_available_worker_count)
+                bulk_logger.info(
+                    f"Not using fork, the environment variable PF_WORKER_COUNT is not set. Calculate the estimated "
+                    f"available worker count based on the current system available memory and process memory. Take the "
+                    f"minimum value among the calculated value: {estimated_available_worker_count}, "
+                    f"the default value for worker_count: {self._worker_count}, and the row count: "
+                    f"{self._nlines}. Process count: {self._n_process}")
```

Comment:
Shall we print such detailed information in one log message?
Can we separate them apart? Like:

* "Using fork to create new processes."
* "The PF_WORKER_COUNT environment is not set or invalid. Calculate the worker count based on the currently memory usage."
* Calculated process count ({self._n_process}) by taking the minimum value among estimated process count ({estmated_available_worker_count}), the row count ({self._nlines}) and the default worker count ({self._worker_count}).

Reply:
fixed